build-logic/common-plugins/src/main/kotlin/org.xtclang.build.java.gradle.kts

```diff
@@ -0,0 +1,129 @@
+import XdkBuildLogic.Companion.DEFAULT_JAVA_BYTECODE_VERSION
+import org.gradle.api.tasks.testing.logging.TestLogEvent.STANDARD_OUT
+import org.gradle.api.tasks.testing.logging.TestLogEvent.STANDARD_ERROR
+import org.gradle.api.tasks.testing.logging.TestLogEvent.SKIPPED
+import org.gradle.api.tasks.testing.logging.TestLogEvent.STARTED
+import org.gradle.api.tasks.testing.logging.TestLogEvent.PASSED
+import org.gradle.api.tasks.testing.logging.TestLogEvent.FAILED
+
+import java.nio.charset.StandardCharsets.UTF_8
+
+plugins {
+    id("org.xtclang.build.xdk.versioning")
+    java
+}
+
+private val pprefix = "org.xtclang.java"
+private val lintProperty = "$pprefix.lint"
+
+private val jdkVersion: Provider<Int> = provider {
+    getXdkPropertyInt("$pprefix.jdk", DEFAULT_JAVA_BYTECODE_VERSION)
+}
+
+java {
+    toolchain {
+        val xdkJavaVersion = JavaLanguageVersion.of(jdkVersion.get())
+        val buildProcessJavaVersion = JavaLanguageVersion.of(JavaVersion.current().majorVersion.toInt())
+        if (!buildProcessJavaVersion.canCompileOrRun(xdkJavaVersion)) {
+            throw buildException("Error in Java Toolchain config. The builder can't compile requested Java version: $xdkJavaVersion")
+        }
+        logger.info("$prefix Java Toolchain config; binary format version: 'JDK $xdkJavaVersion' (build process version: 'JDK $buildProcessJavaVersion')")
+        languageVersion.set(xdkJavaVersion)
+    }
+}
+
+testing {
+    suites {
+        @Suppress("UnstableApiUsage") val test by getting(JvmTestSuite::class) {
+            useJUnitJupiter()
+        }
+    }
+}
+
+tasks.withType<JavaExec>().configureEach {
+    inputs.property("jdkVersion", jdkVersion);
+    logger.info("$prefix Configuring JavaExec task $name from toolchain (Java version: ${java.toolchain.languageVersion})")
+    javaLauncher.set(javaToolchains.launcherFor(java.toolchain))
+    if (enablePreview()) {
+        jvmArgs("--enable-preview")
+    }
+    doLast {
+        logger.info("$prefix JVM arguments: $jvmArgs")
+    }
+}
+
+val checkWarnings by tasks.registering {
+    if (!getXdkPropertyBoolean(lintProperty, false)) {
+        val lintPropertyHasValue = isXdkPropertySet(lintProperty)
+        logger.warn("$prefix WARNING: Java warnings are ${if (lintPropertyHasValue) "explicitly" else ""} for project.!")
```

Comment:
Looks like the produced warning is missing a noun:

> WARNING: Java warnings are explicitly for project.!

Did you mean it to say "explicitely turned off"? Also, I think the dot is unnecessary (or an exclamation mark)